core/src/banking_stage.rs

```diff
@@ -1760,31 +1787,60 @@ impl BankingStage {
         Self::filter_valid_transaction_indexes(&results, transaction_to_packet_indexes)
     }
 
+    fn filter_processed_packets<'a, F>(
+        retryable_transaction_indexes: impl Iterator<Item = &'a usize>,
+        mut f: F,
+    ) where
+        F: FnMut(usize, usize),
+    {
+        let mut prev_retryable_index = 0;
+        for (i, retryable_index) in retryable_transaction_indexes.enumerate() {
+            let start = if i == 0 { 0 } else { prev_retryable_index + 1 };
```

Comment:
I think you could've kept the cleaner `.window()`-based version, just needed to `.enumerate()` then do something like
```
let start = range[0] + i.min(1)
```